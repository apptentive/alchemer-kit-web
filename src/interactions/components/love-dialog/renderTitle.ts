/**
 * Renders the Love Dialog header (title).
 *
 * @param {string} titleText - The text title to render.
 * @returns {HTMLDivElement} - The title container DOM Node.
 */
const renderTitle = (titleText: string) => {
  const titleElement = document.createElement('div');
  titleElement.className = 'apptentive-love-dialog-title';

  const headerElement = document.createElement('h1');
  headerElement.textContent = titleText || '';

  titleElement.append(headerElement);

  return titleElement;
};

export default renderTitle;
