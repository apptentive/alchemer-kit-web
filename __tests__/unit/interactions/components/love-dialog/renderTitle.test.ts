import renderTitle from '../../../../../src/interactions/components/love-dialog/renderTitle';

const _titleText = 'Title';

describe('renderTitle', () => {
  test('properly renders title', () => {
    const output = renderTitle(_titleText);

    expect(output).toMatchSnapshot();
  });

  test('properly renders title when no text is passed in', () => {
    // @ts-expect-error test for passing invalid parameters
    const output = renderTitle();

    expect(output).toMatchSnapshot();
  });
});
