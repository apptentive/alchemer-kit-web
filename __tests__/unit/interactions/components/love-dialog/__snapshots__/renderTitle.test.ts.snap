// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderTitle properly renders title 1`] = `
<div
  class="apptentive-love-dialog-title"
>
  <h1>
    Title
  </h1>
</div>
`;

exports[`renderTitle properly renders title when no text is passed in 1`] = `
<div
  class="apptentive-love-dialog-title"
>
  <h1 />
</div>
`;
